packages/site/plugins/react-router/browser.api.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable react/display-name */
+/* eslint-disable react-hooks/rules-of-hooks */
+
+import React from 'react';
+import { useBasepath, useStaticInfo } from 'react-static';
+import { BrowserRouter, StaticRouter, withRouter } from 'react-router-dom';
+
+const Location = withRouter(({ children, location }) => children(location));
+
+const ReactRouterPlugin = ({ RouterProps: userRouterProps = {} }) => ({
+  Root: PreviousRoot => ({ children }) => {
+    const routerProps = { basename: useBasepath() || '' };
+    if (routerProps.basename) routerProps.basename = `/${routerProps.basename}`;
+    const staticInfo = useStaticInfo();
+
+    // Test for document to detect the node stage
+    let Router;
+    if (typeof document !== 'undefined') {
+      // NOTE: React Router is inconsistent in how it handles base paths
+      // This will need a trailing slash while the StaticRouter does not
+      if (routerProps.basename) routerProps.basename += '/';
```

Comment:
if you want to see something wild, look no more 😆

Reply:
Yea this is bonkers